### DISCUSSION TITLE
help to use the analyzer

### DISCUSSION THREAD

**Discussion by ZzzzHeJ** [nautechsystems/nautilus_trader#2122] [Q&A]
I want to use analysis tools to draw daily revenue curves and statistics Sharp. I set timer and add_return every day, but the returns will reset when the backtest is over. What's the right thing to do please.
```python
            venue_positions = []
            venue_currencies = set()
            for position in positions:
                if position.instrument_id.venue == venue.id:
                    venue_positions.append(position)
                    venue_currencies.add(position.quote_currency)
                    if position.base_currency is not None:
                        venue_currencies.add(position.base_currency)

            # Calculate statistics
            self._kernel.portfolio.analyzer.calculate_statistics(account, venue_positions)
```

**Comment by cjdsellers** [comment]
Hi @ZzzzHeJ

Thanks for reaching out.

All data remains available until `engine.reset()` is called on the engine.

**Reply by ZzzzHeJ** [reply]
I just run 
```python
node = BacktestNode(configs=[config])

result = node.run()
```
it seems like will auto call engine.reset() in the run().
please tell me what should I do, and the best practice to get the daliy return

**Reply by cjdsellers** [reply]
Hi @ZzzzHeJ 

You could try passing a config with [dispose_on_completion](https://github.com/nautechsystems/nautilus_trader/blob/develop/nautilus_trader/backtest/config.py#L320) of False, and see if you can access the data from the analyzer as you expect.

**Reply by ZzzzHeJ** [reply]
Thanks, I will try it, by the way, I found a bug in the analyzer
https://github.com/nautechsystems/nautilus_trader/blob/b65c810fa3d854de84f2ec8ff5f9c2008eab0b9c/nautilus_trader/analysis/analyzer.py#L146
the `sort_index()` is not effect the self._return, maybe use self._return = self._return .sort_index()?

**Reply by ZzzzHeJ** [reply]
Unfortunately, it doesn't work, because the `analyzer._returns` will be reset in the `calculate_statistics`
https://github.com/nautechsystems/nautilus_trader/blob/b65c810fa3d854de84f2ec8ff5f9c2008eab0b9c/nautilus_trader/analysis/analyzer.py#L128-L146

This is the call stack analyzer.calculate_statistics<-engine._log_post_run<-engine.end, the `end()` will call after `_run()` with only `streaming` judgment, which is `True` in the `_run_oneshot` mode


https://github.com/nautechsystems/nautilus_trader/blob/4535ca41cef0d44c8040a4182e5bd9720578bf1a/nautilus_trader/backtest/engine.pyx#L909-L955

**Reply by ZzzzHeJ** [reply]
Comment out this code to make it work？

https://github.com/nautechsystems/nautilus_trader/blob/b65c810fa3d854de84f2ec8ff5f9c2008eab0b9c/nautilus_trader/analysis/analyzer.py#L143
